modules/prefmanager.js

```diff
@@ -127,6 +127,7 @@ GM_PrefManager.prototype.remove = function(prefName) {
  * call a function whenever the named preference subtree changes
  */
 GM_PrefManager.prototype.watch = function(prefName, watcher) {
+  if (this.observers[watcher]) return;
```

Comment:
It's not unreasonable to write a watcher that is capable of observing several preferences, is it not?